go/vt/servenv/mysql.go

```diff
@@ -16,9 +16,49 @@ limitations under the License.
 
 package servenv
 
-import "flag"
+import (
+	"github.com/spf13/pflag"
+)
 
-// MySQLServerVersion is what Vitess will present as it's version during the connection handshake,
+// mySQLServerVersion is what Vitess will present as it's version during the connection handshake,
 // and as the value to the @@version system variable. If nothing is provided, Vitess will report itself as
 // a specific MySQL version with the vitess version appended to it
-var MySQLServerVersion = flag.String("mysql_server_version", "", "MySQL server version to advertise.")
+var mySQLServerVersion string
+
+// RegisterMySQLServerFlags installs the flags needed to specify or expose a
+// particular MySQL server version from Vitess.
+func RegisterMySQLServerFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&mySQLServerVersion, "mysql_server_version", mySQLServerVersion, "MySQL server version to advertise.")
+}
+
+// MySQLServerVersion returns the value of the `--mysql_server_version` flag.
+func MySQLServerVersion() string {
+	return mySQLServerVersion
+}
+
+// SetMySQLServerVersionForTest sets the value of the `--mysql_server_version`
+// flag. It is intended for use in tests that require a specific MySQL server
+// version (for example, collations) that cannot specify that via the command
+// line.
+func SetMySQLServerVersionForTest(version string) {
+	mySQLServerVersion = version
+}
+
+func init() {
+	for _, cmd := range []string{
+		"mysqlctl",
+		"mysqlctld",
+		"vtbackup",
+		"vtcombo",
+		"vtctl",
+		"vtctld",
+		"vtctldclient",
```

Comment:
I am unsure about the vtclt binaries as well.

Reply:
`vtctl` uses `sqlparser.Parse` for a bunch of functionality, among them:

- `vtctl OnlineDDL`: https://github.com/vitessio/vitess/blob/main/go/vt/vtctl/vtctl.go#L3171-L3177
- `vtctl ApplyVSchema`: https://github.com/vitessio/vitess/blob/main/go/vt/vtctl/vtctl.go#L3462
- some of the workflow/vexec/vreplication queries use it: https://github.com/vitessio/vitess/blob/main/go/vt/vtctl/workflow/vexec/vexec.go#L130

Reply:
Yes, you are correct! I totally forgot about the OnlineDDL using it. I thought that we deprecated the execute as VT_dba, etc sort of functions, so I thought it might not be required!